burnman/mademember.py

```diff
@@ -0,0 +1,290 @@
+        
+    @material_property
+    def isothermal_compressibility(self):
+        """
+        Returns isothermal compressibility of the solid solution (or inverse isothermal bulk modulus) [1/Pa]
+        Aliased with self.K_T
+        """
+        return 1. / self.isothermal_bulk_modulus
+
+    @material_property
+    def adiabatic_compressibility(self):
+        """
+        Returns adiabatic compressibility of the solid solution (or inverse adiabatic bulk modulus) [1/Pa]
+        Aliased with self.K_S
+        """
+        return 1. / self.adiabatic_bulk_modulus
+
+    @material_property
+    def p_wave_velocity(self):
+        """
+        Returns P wave speed of the solid solution [m/s]
+        Aliased with self.v_p
+        """
+        return np.sqrt((self.adiabatic_bulk_modulus + 4. / 3. *
+                        self.shear_modulus) / self.density)
+
+    @material_property
+    def bulk_sound_velocity(self):
+        """
+        Returns bulk sound speed of the solid solution [m/s]
+        Aliased with self.v_phi
+        """
+        return np.sqrt(self.adiabatic_bulk_modulus / self.density)
+
+    @material_property
+    def shear_wave_velocity(self):
+        """
+        Returns shear wave speed of the solid solution [m/s]
+        Aliased with self.v_s
+        """
+        return np.sqrt(self.shear_modulus / self.density)
+
+    @material_property
+    def grueneisen_parameter(self):
+        """
+        Returns grueneisen parameter of the solid solution [unitless]
+        Aliased with self.gr
+        """
+        if self.temperature < 1.e-12:
+            return 0.
+        else:
+            return self.thermal_expansivity * self.isothermal_bulk_modulus \
+                * self.molar_volume / self.heat_capacity_v
+
+    @material_property
+    def heat_capacity_v(self):
+        """
+        Returns heat capacity at constant volume of the solid solution [J/K/mol]
+        Aliased with self.C_v
+        """
+        return self.heat_capacity_p - self.molar_volume * self.temperature \
+            * self.thermal_expansivity * self.thermal_expansivity \
+            * self.isothermal_bulk_modulus
+
+
+def make_endmember(mineral_list, molar_amounts, free_energy_adjustment):
+    """
+    This function makes a new endmember from a linear combination of minerals
+    and a Gibbs free energy adjustment which is linear in
+    pressure and temperature (i.e. a constant energy, entropy and
+    volume adjustment).
+    
+    For example, a crude approximation to a bridgmanite model might be
+    bdg = make_endmember([per, stv], [1.0, 1.0], [-15.e3, 0., 0.])
+    """
+    assert(len(free_energy_adjustment) == 3)
+
+    class mechanical_mixture(SolidSolution):
+        def __init__(self, mineral_list, molar_amounts):
+            self.type = 'mechanical'
+            self.endmembers = [[m, ''] for m in mineral_list]
+            SolidSolution.__init__(self, molar_amounts)
+
+    class made_endmember(MadeMember):
```

Comment:
can't you do
```
x = MadeMember(mechanical_mixture(mineral_list, molar_amounts))
x.property_modifiers = ...
return x
```
this way you don't have to create a new class in the hierarchy that doesn't do anything useful. (This needs the change I mention above).